pandas/core/dtypes/cast.py

```diff
@@ -1652,7 +1652,12 @@ def construct_1d_arraylike_from_scalar(
 
     if isinstance(dtype, ExtensionDtype):
         cls = dtype.construct_array_type()
-        subarr = cls._from_sequence([value] * length, dtype=dtype)
+        if isinstance(dtype, CategoricalDtype):
+            subarr = cls._from_sequence([value] * length, dtype=dtype)
```

Comment:
what happens if you don't special case Categorical?

Reply:
Calling `take` on the empty categorical would error because the categories were not defined. 

I just pushed another commit which first sets the categories. Categorical now has a similar improvement:

```
import pandas as pd

N = 1_000_000

%timeit pd.Series(1, index=range(N), dtype='category')
415 ms ± 91.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)     <- main
4.5 ms ± 1.57 ms per loop (mean ± std. dev. of 7 runs, 100 loops each)  <- PR

Reply:
> Calling take on the empty categorical would error because the categories were not defined.

Hmm i think this might be a problem for other not-fully-initialized dtype objects.  IIRC IntervalDtype without 'closed' set is another one, not sure if we have a way to detect the general case

Reply:
latest update uses `.repeat` per @jreback's suggestion below.